aws.dockerfile

```diff
@@ -0,0 +1,99 @@
+FROM maven:3-jdk-8-alpine as java-build
+
+COPY kinesis-client /opt/sites/rig/kinesis-client
+
+WORKDIR /opt/sites/rig/kinesis-client
+
+# Compile AWS Kinesis Java application
+RUN mvn package
+
+FROM elixir:1.6-alpine as elixir-build
+
+# Install Elixir & Erlang environment dependencies
+RUN mix local.hex --force
+RUN mix local.rebar --force
+RUN apk update && apk add make \
+    gcc \
+    g++
+
+ENV MIX_ENV=prod
+
+WORKDIR /opt/sites/rig
+
+# Copy release config
+COPY version /opt/sites/rig/
+COPY rel /opt/sites/rig/rel/
+COPY vm.args /opt/sites/rig/
+
+# Copy necessary files for dependencies
+COPY mix.exs /opt/sites/rig/
+COPY mix.lock /opt/sites/rig/
+COPY apps/rig/mix.exs /opt/sites/rig/apps/rig/
+COPY apps/rig_api/mix.exs /opt/sites/rig/apps/rig_api/
+COPY apps/rig_auth/mix.exs /opt/sites/rig/apps/rig_auth/
+COPY apps/rig_inbound_gateway/mix.exs /opt/sites/rig/apps/rig_inbound_gateway/
+COPY apps/rig_outbound_gateway/mix.exs /opt/sites/rig/apps/rig_outbound_gateway/
```

Comment:
I thought we've settled on using dockerignore instead of this mount of COPY statements...

Reply:
based on official Java Dockerfile https://github.com/docker-library/openjdk/blob/9a0822673dffd3e5ba66f18a8547aa60faed6d08/8-jdk/alpine/Dockerfile might try if this will suffice

Reply:
tbh, read some articles and `.dockerignore` is not always the best approach, but I'll think about it ...

Reply:
Problem with general `.dockerignore` here is that e.g. if we put tests there, then dockerfile for smoke tests will exclude them as well, so no tests to run. There is a long discussion about it here https://github.com/moby/moby/issues/12886, but no solution to this day. I would not heavily rely on `.dockerignore` till it's solved by guys from Docker.